### PR TITLE
chore: update each cmdlet to include the `.PARAMETER` details

### LINKED ISSUE #23: Update each cmdlet to include the `.PARAMETER` details
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

Update each module cmdlet to include the `.PARAMETER` for each cmdlet parameter.

This will be used to generate the documentation and provide better help.

### Use Case(s)

- WIll be used to generate the documentation.
- Is useful in the cmdlet help.

### Potential Configuration

The `.PARAMETER` for each cmdlet parameter should be similar to the following:

```powershell
        ...
        
        .PARAMETER server
        The FQDN of the SDDC Manager appliance.
        
        .PARAMETER user
        The username to authenticate to SDDC Manager.
        
        .PARAMETER pass
        The password to authenticate to SDDC Manager.
        
        .PARAMETER domain
        The name of the workload domain to retrieve the vCenter Server details from SDDC Manager for the connection object.
        
        ...
```

See `VMware.CloudFoundation.PasswordManagement` as an example.

### References

_No response_

### PR DESCRIPTION
**Pull request by sowjuec**
### Summary

Updated each cmdlet to include the `.PARAMETER` details.

### Type

- [ ] Bugfix
- [ ] Enhancement or Feature
- [ ] Code Style or Formatting
- [X] Documentation
- [ ] Refactoring
- [X] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### Test and Documentation

- [X] Tests have been completed.
- [ ] Documentation has been added or updated.

### Issue References

Closes #23

### CODE REVIEW

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 1]
```diff
@@ -321,6 +360,18 @@ Function Set-MaintenanceMode {
...
+        The FQDN of the Esxi host.
+
+        .PARAMETER user
+        The username to authenticate to Esxi host.
+
+        .PARAMETER pass
+        The password to authenticate to Esxi host.
+
+        .PARAMETER state
+        The state of the MaintenanceMode to be set on Esxi host. Allowed states are "ENABLE" or "DISABLE.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:374]:
```suggestion
        The state of the maintenance mode to be set on ESXi host. Allowed states are "ENABLE" or "DISABLE".
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 2]
```diff
@@ -463,6 +523,21 @@ Function Set-DrsAutomationLevel {
...
+        The username to authenticate to Virtual Center Server.
+
+        .PARAMETER pass
+        The password to authenticate to Virtual Center Server.
+
+        .PARAMETER cluster
+        The name of the cluster on which the DRS automation level settings are to be applied.
+
+        .PARAMETER level
+        The DRS automation level to be set.  The value can be one amongst ("FullyAutomated", "Manual", "PartiallyAutomated", "Disabled").
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:540]:
```suggestion
        The DRS automation level to be set. One of: "FullyAutomated", "Manual", "PartiallyAutomated", "Disabled".
```

File: VMware.CloudFoundation.PowerManagement.psm1 (hunk 1 above)
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:374]:
```suggestion
        The state of the maintenance mode to be set on ESXi host. Allowed states are "ENABLE" or "DISABLE".
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 3]
```diff
@@ -535,6 +610,24 @@ Function Set-VsanClusterPowerStatus {
...
+        The FQDN of the Virtual Center Server.
+
+        .PARAMETER user
+        The username to authenticate to Virtual Center Server.
+
+        .PARAMETER pass
+        The password to authenticate to Virtual Center Server.
+
+        .PARAMETER clustername
+        The name of the VSAN cluster on which the power settings are to be applied.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:624]:
```suggestion
        The name of the vSAN cluster on which the power settings are to be applied.
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 4]
```diff
@@ -535,6 +610,24 @@ Function Set-VsanClusterPowerStatus {
...
+        
+        .PARAMETER mgmt
+        The switch used to ignore powersettings if management domain information is passed.
+
+        .PARAMETER PowerStatus
+        The power state to be set for a given VSAN cluster.  The value can be one amongst ("clusterPoweredOff", "clusterPoweredOn").
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:630]:
```suggestion
        The power state to be set for a given vSAN cluster.  The value can be one amongst ("clusterPoweredOff", "clusterPoweredOn").
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 5]
```diff
@@ -632,6 +725,18 @@ Function Get-poweronVMsOnRemoteDS {
         .EXAMPLE
         Get-poweronVMsOnRemoteDS -server sfo-m01-vc01.sfo.rainpole.io -user administrator@vsphere.local  -Pass VMw@re1! -clustertocheck sfo-m01-cl01
         This example returns the list of VMs that reside on a vSAN HCI Mesh datastore hosted in cluster sfo-m01-cl01
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:727]:
```suggestion
        This example returns the list of VMs that reside on a vSAN HCI Mesh datastore hosted in cluster sfo-m01-cl01.
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 6]
```diff
@@ -962,6 +1119,18 @@ Function Test-VsanHealth {
         .EXAMPLE
         Test-VsanHealth -cluster sfo-m01-cl01 -server sfo-m01-vc01 -user administrator@vsphere.local -pass VMw@re1!
         This example connects to a vCenter Server and checks the state of the vSAN cluster health
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:1121]:
```suggestion
        This example connects to a vCenter Server and checks the state of the vSAN cluster health.
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 7]
```diff
@@ -1066,6 +1235,18 @@ Function Test-VsanObjectResync {
...
+        The FQDN of the Virtual Center server.
+
+        .PARAMETER user
+        The username to authenticate to Virtual Center server.
+
+        .PARAMETER pass
+        The password to authenticate to Virtual Center server.
+
+        .PARAMETER cluster
+        The name of the VSAN cluster on which object resynchronization status has to be checked.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:1249]:
```suggestion
        The name of the vSAN cluster on which object resynchronization status has to be checked.
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 8]
```diff
@@ -1400,6 +1635,24 @@ Function Set-VsphereHA {
...
+        The password to authenticate to Virtual Center server.
+        
+        .PARAMETER cluster
+        The name of the cluster.
+
+        .PARAMETER enableHA
+        The switch to enable high availability.
+
+        .PARAMETER disableHA
+        The switch to disable high availability.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:1655]:
```suggestion
        The switch to disable vSphere High Availability.
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 9]
```diff
@@ -2040,6 +2411,6 @@ Function Debug-CatchWriterForPowerManagement {
     Write-PowerManagementLogMessage -message " ERROR Message: $errorMessage" -Colour Red
     Write-Error -Message $errorMessage
 }
-Export-ModuleMember -Function Debug-CatchWriterForPowerManagement
+#Export-ModuleMember -Function Debug-CatchWriterForPowerManagement
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:2414]: Can we just remove vs comment out?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
 
+## [v1.1.1](UnReleased)
+
+### Enhancements
+
+- Updated each cmdlet to include the .PARAMETER details
```
Comment (tenthirtyam) [inline CHANGELOG.md:7]:
```suggestion
- Updated each cmdlet to include the `.PARAMETER` details. [GH-27](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-power-management/pull/27)
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,11 @@
 # CHANGELOG
```
Comment (tenthirtyam) [inline CHANGELOG.md:1]:
```suggestion
# Release History
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,11 @@
...
 
+## [v1.1.1](UnReleased)
```
Comment (tenthirtyam) [inline CHANGELOG.md:3]:
```suggestion
## v1.1.1 (Unreleased)

> Release Date: Unreleased
```

File: VMware.CloudFoundation.PowerManagement.psm1 (hunk 9 above)
Comment (sowjuec) [reply VMware.CloudFoundation.PowerManagement.psm1:2414]: yeah

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 13]
```diff
@@ -463,6 +523,21 @@ Function Set-DrsAutomationLevel {
...
+        The username to authenticate to vCenter Server.
+
+        .PARAMETER pass
+        The password to authenticate to vCenter Server.
+
+        .PARAMETER cluster
+        The name of the cluster on which the DRS automation level settings are to be applied.
+
+        .PARAMETER level
+        The DRS automation level to be set. One of: "FullyAutomated", "Manual", "PartiallyAutomated", "Disabled".
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:540]:
```suggestion
        The DRS automation level to be set. The value can be one amongst ("FullyAutomated", "Manual", "PartiallyAutomated", "Disabled").
```

Comment (tenthirtyam) [reply VMware.CloudFoundation.PowerManagement.psm1:540]: Sorry!

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 14]
```diff
@@ -535,6 +610,24 @@ Function Set-VsanClusterPowerStatus {
...
+        The password to authenticate to vCenter Server.
+
+        .PARAMETER clustername
+        The name of the vSAN cluster on which the power settings are to be applied.
+        
+        .PARAMETER mgmt
+        The switch used to ignore power settings if management domain information is passed.
+
+        .PARAMETER PowerStatus
+        The power state to be set for a given vSAN cluster.  The value can be one amongst ("clusterPoweredOff", "clusterPoweredOn").
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:630]:
```suggestion
        The power state to be set for a given vSAN cluster. The value can be one amongst ("clusterPoweredOff", "clusterPoweredOn").
```

File: VMware.CloudFoundation.PowerManagement.psm1 [hunk 15]
```diff
@@ -854,7 +986,22 @@ Function Invoke-EsxCommand {
...
+        The username to authenticate to ESXi host.
+
+        .PARAMETER pass
+        The password to authenticate to ESXi host.
+
+        .PARAMETER cmd
+        The command to be exectued on the ESXi host.
+
+        .PARAMETER expected
+        The expected output to be compared against output got from the command execution.
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.PowerManagement.psm1:1003]:
```suggestion
        The expected output to be compared against output returned from the command execution.
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
Very minor changes request. Will approve and merge once addressed.

**Comment by tenthirtyam** [comment]
I'll need to rebase to address the merge conflicts. I can do that after the requested changes are made.

**Review by tenthirtyam — APPROVED** [review]

**Pull request merged**